tools/integration_tests/gzip/gzip_test.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
```

Comment:
please add unit tests in this PR and put the integration tests in seperate PR

Reply:
Done. Added in file_test.go .